src/core/proc_maps/mod.rs

```diff
@@ -0,0 +1,18 @@
+#[cfg(target_os = "macos")]
+pub mod mac_maps;
+#[cfg(target_os = "macos")]
+pub use self::mac_maps::{get_process_maps, MapRange, Pid};
+
+#[cfg(target_os = "linux")]
+pub mod linux_maps;
+#[cfg(target_os = "linux")]
+pub use self::linux_maps::{get_process_maps, MapRange, Pid};
+
+fn map_contain_addr(map: &MapRange, addr: usize) -> bool {
+    let start = map.start();
+    (addr > start) && (addr < (start + map.size()))
+}
+
+pub fn maps_contain_addr(addr: usize, maps: &[MapRange]) -> bool {
+    maps.iter().any({ |map| map_contain_addr(map, addr) })
+}
```

Comment:
I find this a bit confusing because the `maps_contain_addr` code is actually only used on Linux, and so it seems misleading today to write a generic version of it. 

here's a commit that might clarify things -- it takes all the usages of `maps_contain_addr` and marks them to only compile on Linux. I think that code will still compile on Mac. https://github.com/rbspy/rbspy/commit/e3f52831f369f48b30fa7976b0a0c6b69021dfbe

Reply:
I agree with you that having this generic functionality isn’t needed yet - but the cool thing is that that the ‘could_be_thread’ and ’is_maybe_thread’ functions will work on OSX without any further changes like this. 

Like, we could support stripped binaries on OSX pretty easily here - just by extracting the __bss section address from the mach-o ruby binaries and then using the same code as on linux for scanning for the currentthread addresses.

Your change doesn't quite compile on OSX, I think something like this would work though if you want go this route instead: https://github.com/benfred/rbspy/commit/749938a81239f03301e8abb41ddd04200dcbe587